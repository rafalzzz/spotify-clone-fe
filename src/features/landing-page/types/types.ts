import { EAlbumKeys, TAlbum } from '@/types/album';
import { TArtist } from '@/types/artist';
import { TSongItem } from '@/types/components';
import { TMusicTrack } from '@/types/music-track';

export type TArtistSection = {
  artists: TArtist[];
};

export type TLoveAlbumsSection = {
  albums: TAlbum[];
};

export type TLoveSongsSection = { songs: TMusicTrack[] };

export type TLoveSongsSectionItem = {
  song: TMusicTrack;
  isPlaying: boolean;
  isActive: boolean;
};

export type TMainContent = {
  songs: TMusicTrack[];
  albums: TAlbum[];
  artists: TArtist[];
};

export type TUseLandingPageProps = TMainContent;

export type TUseLoveSongsSectionProps = {
  isPlaying: boolean;
  currentPlayedSong: TSongItem;
  handleOnClick: (songItem: TSongItem) => void;
};

export type TAlbumItem = Pick<
  TAlbum,
  | EAlbumKeys.COLLECTION_ID
  | EAlbumKeys.COLLECTION_NAME
  | EAlbumKeys.ARTWORK_URL_60
  | EAlbumKeys.ARTIST_NAME
  | EAlbumKeys.RELEASE_DATE
>;

export type TLoveAlbumsSectionItem = {
  album: TAlbumItem;
  isPlaying: boolean;
  isActive: boolean;
};

export type TUseLoveSongsItemContextMenu = {
  song: TMusicTrack;
  copytoClipboard: (text: string) => void;
};
