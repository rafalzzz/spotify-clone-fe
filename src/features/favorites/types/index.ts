import { TSongItem } from '@/types/components';

export type TContextMenuCell = {
  record: TSongItem;
};
