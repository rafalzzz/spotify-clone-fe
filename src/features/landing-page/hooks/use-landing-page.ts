import { fetchMusicData } from '@/server-actions/fetch-music-data';

export const useLandingPage = async () => {
  const getLoveSongs = fetchMusicData({
    term: 'love',
    entity: 'musicTrack',
    limit: 15,
    errorMessage: 'Failed to fetch songs',
  });

  const getLoveAlbums = fetchMusicData({
    term: 'love',
    entity: 'album',
    limit: 15,
    errorMessage: 'Failed to fetch albums',
  });

  const [songs, albums] = await Promise.all([getLoveSongs, getLoveAlbums]);

  return { songs, albums };
};
