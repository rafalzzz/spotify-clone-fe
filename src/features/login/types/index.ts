import { INITIAL_VALUES } from '@/login/consts';

export type TLoginForm = typeof INITIAL_VALUES;

export enum ELoginFormKeys {
  LOGIN = 'login',
  PASSWORD = 'password',
  REMEMBER_ME = 'rememberMe',
}
