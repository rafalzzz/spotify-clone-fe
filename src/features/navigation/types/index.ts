export enum EHistoryActionKind {
  UNDO_BUTTON = 'UNDO_BUTTON',
  REDO_BUTTON = 'REDO_BUTTON',
  ADD_PATH = 'ADD_PATH',
  UPDATE_PATHS = 'UPDATE_PATHS',
  UPDATE_PATH_INDEX = 'UPDATE_PATH_INDEX',
}

export type TAddPath = { pathname: string };
export type TUpdatePathIndex = { value: number };

export type TCountAction = {
  type: EHistoryActionKind;
  payload?: TAddPath | TUpdatePathIndex;
};

export type TCountState = {
  allPaths: string[];
  currentPathIndex: number;
};
