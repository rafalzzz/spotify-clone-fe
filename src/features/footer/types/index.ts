import { ChangeEvent, Dispatch, RefObject, SetStateAction, SyntheticEvent } from 'react';

import { TSongItem } from '@/types/components';

export type TMusicPlayerContext = {
  ref: RefObject<HTMLAudioElement>;
  isShuffle: boolean;
  isLoop: boolean;
  setIsShuffle: Dispatch<React.SetStateAction<boolean>>;
  setIsLoop: Dispatch<React.SetStateAction<boolean>>;
};

export type TProgressBar = {
  value: number;
  minValue: number;
  maxValue: number;
  handleChange: (event: ChangeEvent<HTMLInputElement>) => void;
  handleStartChange?: () => void;
  handleEndChange?: () => void;
};

export const enum ETextAlign {
  START,
  END,
}

export type TDuration = {
  seconds: number;
  textAlign?: ETextAlign;
  disabled?: boolean;
  isReversedTime?: boolean;
  onClick?: () => void;
};

export type TUseIsTextOverflowing = {
  currentSong: TSongItem;
};

export type TUseIsTextOverflowingProps = {
  ref: RefObject<HTMLDivElement>;
  isTextOverflowing: boolean;
};

export type TAudio = {
  setCurrentTime: Dispatch<React.SetStateAction<number>>;
};

export type TUseAudio = TAudio;

export type TUseAudioProps = {
  ref: RefObject<HTMLAudioElement>;
  currentSong: TSongItem;
  isPlaying: boolean;
  isLoop: boolean;
  onLoadedMetadata: ({ target }: SyntheticEvent<HTMLAudioElement>) => void;
  onTimeUpdate: ({ target }: SyntheticEvent<HTMLAudioElement>) => void;
  onEnded: () => void;
};

export type TUsePlayerButtonsProps = {
  isPlaying: boolean;
};

export type TUseMusicProgressBarProps = {
  duration: number;
  currentTime: number;
  temporaryTime: number | null;
  durationValue: number;
  isReversedTime: boolean;
  setCurrentTime: Dispatch<SetStateAction<number>>;
  handleStartChange: () => void;
  handleChange: (event: ChangeEvent<HTMLInputElement>) => void;
  handleEndChange: () => void;
  onClick: () => void;
};

export type TUseSoundProgressBar = {
  isMuted: boolean;
  volume: number;
  toggleMuted: () => void;
  handleChange: ({ target: { valueAsNumber } }: ChangeEvent<HTMLInputElement>) => void;
};
