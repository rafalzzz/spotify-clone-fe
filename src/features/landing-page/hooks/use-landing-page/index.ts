import { fetchMusicData } from '@/server-actions/fetch-music-data';

import { TAlbum } from '@/types/album';
import { TArtist } from '@/types/artist';
import { TMusicTrack } from '@/types/music-track';

import { TUseLandingPageProps } from '../../types';

export const useLandingPage = async (): Promise<TUseLandingPageProps> => {
  const loveSongsPromise = fetchMusicData<TMusicTrack[]>({
    term: 'love',
    entity: 'musicTrack',
    limit: 15,
    errorMessage: 'Failed to fetch songs',
  });

  const loveAlbumsPromise = fetchMusicData<TAlbum[]>({
    term: 'love',
    entity: 'album',
    limit: 15,
    errorMessage: 'Failed to fetch albums',
  });

  const discJockeysPromise = fetchMusicData<TArtist[]>({
    term: 'dj',
    entity: 'musicArtist',
    limit: 15,
    errorMessage: 'Failed to fetch Disc Jokecys',
  });

  const [songs, albums, artists] = await Promise.all([
    loveSongsPromise,
    loveAlbumsPromise,
    discJockeysPromise,
  ]);

  return { songs, albums, artists };
};
