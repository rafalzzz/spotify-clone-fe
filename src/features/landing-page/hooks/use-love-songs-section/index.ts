'use client';
import { useCallback, useMemo } from 'react';

import { TUseLoveSongsSectionProps } from '@/landing-page/types/types';

import { useMusicPlayerStore } from '@/store/music-player';

import { EMusicTrackKeys } from '@/types/music-track';
import { TSongItem } from '@/types/store';

export const useLoveSongsSection = (): TUseLoveSongsSectionProps => {
  const { isPlaying, activeIndex, songsList, changeSong, togglePlay } = useMusicPlayerStore();

  const currentPlayedSong = useMemo(() => songsList[activeIndex], [activeIndex, songsList]);

  const handleOnClick = useCallback(
    (songItem: TSongItem) => {
      const trackIsAlreadyPlayed =
        songItem[EMusicTrackKeys.PREVIEW_URL] === currentPlayedSong?.[EMusicTrackKeys.PREVIEW_URL];

      if (trackIsAlreadyPlayed) {
        return togglePlay();
      }

      changeSong({ activeIndex: 0, songs: [songItem] });
    },
    [currentPlayedSong, changeSong, togglePlay],
  );

  return { isPlaying, currentPlayedSong, handleOnClick };
};
