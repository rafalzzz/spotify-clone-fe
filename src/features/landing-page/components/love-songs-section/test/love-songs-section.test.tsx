import { render } from '@testing-library/react';

import { useCalculateSectionItemsAmount } from '@/hooks/use-calculate-section-items-amount';

import { MusicTrack } from '@/types/music-track';

import { LoveSongsSection } from '..';

jest.mock('@/hooks/use-calculate-section-items-amount', () => ({
  useCalculateSectionItemsAmount: jest.fn(),
}));

const mockSongs = [
  {
    trackId: 1,
    artworkUrl100: '/some-image-url1.jpg',
    trackName: 'TestTrack',
    artistName: 'Test_Artist',
  },
  {
    trackId: 2,
    artworkUrl100: '/some-image-url1.jpg',
    trackName: 'TestTrack',
    artistName: 'Test_Artist',
  },
  {
    trackId: 3,
    artworkUrl100: '/some-image-url1.jpg',
    trackName: 'TestTrack',
    artistName: 'Test_Artist',
  },
  {
    trackId: 4,
    artworkUrl100: '/some-image-url1.jpg',
    trackName: 'TestTrack',
    artistName: 'Test_Artist',
  },
  {
    trackId: 5,
    artworkUrl100: '/some-image-url1.jpg',
    trackName: 'TestTrack',
    artistName: 'Test_Artist',
  },
  {
    trackId: 6,
    artworkUrl100: '/some-image-url1.jpg',
    trackName: 'TestTrack',
    artistName: 'Test_Artist',
  },
];

const renderLoveSongsSection = () => render(<LoveSongsSection songs={mockSongs as MusicTrack[]} />);

describe('LoveSongsSection', () => {
  beforeAll(() => {
    (useCalculateSectionItemsAmount as jest.Mock).mockReturnValue({
      elementRef: jest.fn(),
      sectionItemsCount: 3,
    });
  });

  it('render component without error', () => {
    const screen = renderLoveSongsSection();
    expect(screen).toMatchSnapshot();
  });

  it('displays the correct number of songs', () => {
    const screen = renderLoveSongsSection();

    expect(screen.getAllByRole('listitem').length).toBe(3);
  });
});
