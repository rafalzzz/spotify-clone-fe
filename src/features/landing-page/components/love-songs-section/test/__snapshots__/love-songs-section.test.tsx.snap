// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoveSongsSection render component without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="custom-section"
      >
        <div
          class="custom-section__header"
        >
          <header>
            <a
              class="custom-section__header__title"
              data-testid="custom-section-header"
              href="/love-songs"
            >
              <h2>
                Love songs
              </h2>
            </a>
          </header>
          <a
            class="custom-section__header__redirection"
            data-testid="custom-section-show-all-items"
            href="/love-songs"
          >
            Show all
          </a>
        </div>
        <ul
          class="custom-section__items"
        >
          <li>
            <div
              class="custom-section-item"
            >
              <div
                class="custom-section-item__image"
              >
                <img
                  alt="image"
                  data-nimg="1"
                  decoding="async"
                  height="100"
                  loading="lazy"
                  src="/_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75"
                  srcset="/_next/image?url=%2Fsome-image-url1.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75 2x"
                  style="color: transparent; height: 100%; width: 100%;"
                  width="100"
                />
                <button
                  class="custom-section-item__play-button"
                  data-testid="custom-section-item-play-button"
                >
                  <span
                    aria-label="caret-right"
                    class="anticon anticon-caret-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="caret-right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <div
                class="music-track-informations"
              >
                <div
                  data-testid="music-track-informations-track-name-tooltip"
                  placement="right"
                >
                  <h4
                    class="music-track-informations__text"
                    data-testid="music-track-informations-track-name"
                  >
                    TestTrack
                  </h4>
                  <span>
                    TestTrack
                  </span>
                </div>
                <div
                  data-testid="music-track-informations-artist-name-tooltip"
                  placement="right"
                >
                  <a
                    class="music-track-informations__text music-track-informations__artist"
                    href="/artist/test_artist"
                  >
                    <span
                      data-testid="music-track-informations-artist-name"
                    >
                      Test_Artist
                    </span>
                  </a>
                  <span>
                    Test_Artist
                  </span>
                </div>
              </div>
            </div>
          </li>
          <li>
            <div
              class="custom-section-item"
            >
              <div
                class="custom-section-item__image"
              >
                <img
                  alt="image"
                  data-nimg="1"
                  decoding="async"
                  height="100"
                  loading="lazy"
                  src="/_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75"
                  srcset="/_next/image?url=%2Fsome-image-url1.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75 2x"
                  style="color: transparent; height: 100%; width: 100%;"
                  width="100"
                />
                <button
                  class="custom-section-item__play-button"
                  data-testid="custom-section-item-play-button"
                >
                  <span
                    aria-label="caret-right"
                    class="anticon anticon-caret-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="caret-right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <div
                class="music-track-informations"
              >
                <div
                  data-testid="music-track-informations-track-name-tooltip"
                  placement="right"
                >
                  <h4
                    class="music-track-informations__text"
                    data-testid="music-track-informations-track-name"
                  >
                    TestTrack
                  </h4>
                  <span>
                    TestTrack
                  </span>
                </div>
                <div
                  data-testid="music-track-informations-artist-name-tooltip"
                  placement="right"
                >
                  <a
                    class="music-track-informations__text music-track-informations__artist"
                    href="/artist/test_artist"
                  >
                    <span
                      data-testid="music-track-informations-artist-name"
                    >
                      Test_Artist
                    </span>
                  </a>
                  <span>
                    Test_Artist
                  </span>
                </div>
              </div>
            </div>
          </li>
          <li>
            <div
              class="custom-section-item"
            >
              <div
                class="custom-section-item__image"
              >
                <img
                  alt="image"
                  data-nimg="1"
                  decoding="async"
                  height="100"
                  loading="lazy"
                  src="/_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75"
                  srcset="/_next/image?url=%2Fsome-image-url1.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75 2x"
                  style="color: transparent; height: 100%; width: 100%;"
                  width="100"
                />
                <button
                  class="custom-section-item__play-button"
                  data-testid="custom-section-item-play-button"
                >
                  <span
                    aria-label="caret-right"
                    class="anticon anticon-caret-right"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      data-icon="caret-right"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="0 0 1024 1024"
                      width="1em"
                    >
                      <path
                        d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
                      />
                    </svg>
                  </span>
                </button>
              </div>
              <div
                class="music-track-informations"
              >
                <div
                  data-testid="music-track-informations-track-name-tooltip"
                  placement="right"
                >
                  <h4
                    class="music-track-informations__text"
                    data-testid="music-track-informations-track-name"
                  >
                    TestTrack
                  </h4>
                  <span>
                    TestTrack
                  </span>
                </div>
                <div
                  data-testid="music-track-informations-artist-name-tooltip"
                  placement="right"
                >
                  <a
                    class="music-track-informations__text music-track-informations__artist"
                    href="/artist/test_artist"
                  >
                    <span
                      data-testid="music-track-informations-artist-name"
                    >
                      Test_Artist
                    </span>
                  </a>
                  <span>
                    Test_Artist
                  </span>
                </div>
              </div>
            </div>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="custom-section"
    >
      <div
        class="custom-section__header"
      >
        <header>
          <a
            class="custom-section__header__title"
            data-testid="custom-section-header"
            href="/love-songs"
          >
            <h2>
              Love songs
            </h2>
          </a>
        </header>
        <a
          class="custom-section__header__redirection"
          data-testid="custom-section-show-all-items"
          href="/love-songs"
        >
          Show all
        </a>
      </div>
      <ul
        class="custom-section__items"
      >
        <li>
          <div
            class="custom-section-item"
          >
            <div
              class="custom-section-item__image"
            >
              <img
                alt="image"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fsome-image-url1.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75 2x"
                style="color: transparent; height: 100%; width: 100%;"
                width="100"
              />
              <button
                class="custom-section-item__play-button"
                data-testid="custom-section-item-play-button"
              >
                <span
                  aria-label="caret-right"
                  class="anticon anticon-caret-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="caret-right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="music-track-informations"
            >
              <div
                data-testid="music-track-informations-track-name-tooltip"
                placement="right"
              >
                <h4
                  class="music-track-informations__text"
                  data-testid="music-track-informations-track-name"
                >
                  TestTrack
                </h4>
                <span>
                  TestTrack
                </span>
              </div>
              <div
                data-testid="music-track-informations-artist-name-tooltip"
                placement="right"
              >
                <a
                  class="music-track-informations__text music-track-informations__artist"
                  href="/artist/test_artist"
                >
                  <span
                    data-testid="music-track-informations-artist-name"
                  >
                    Test_Artist
                  </span>
                </a>
                <span>
                  Test_Artist
                </span>
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            class="custom-section-item"
          >
            <div
              class="custom-section-item__image"
            >
              <img
                alt="image"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fsome-image-url1.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75 2x"
                style="color: transparent; height: 100%; width: 100%;"
                width="100"
              />
              <button
                class="custom-section-item__play-button"
                data-testid="custom-section-item-play-button"
              >
                <span
                  aria-label="caret-right"
                  class="anticon anticon-caret-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="caret-right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="music-track-informations"
            >
              <div
                data-testid="music-track-informations-track-name-tooltip"
                placement="right"
              >
                <h4
                  class="music-track-informations__text"
                  data-testid="music-track-informations-track-name"
                >
                  TestTrack
                </h4>
                <span>
                  TestTrack
                </span>
              </div>
              <div
                data-testid="music-track-informations-artist-name-tooltip"
                placement="right"
              >
                <a
                  class="music-track-informations__text music-track-informations__artist"
                  href="/artist/test_artist"
                >
                  <span
                    data-testid="music-track-informations-artist-name"
                  >
                    Test_Artist
                  </span>
                </a>
                <span>
                  Test_Artist
                </span>
              </div>
            </div>
          </div>
        </li>
        <li>
          <div
            class="custom-section-item"
          >
            <div
              class="custom-section-item__image"
            >
              <img
                alt="image"
                data-nimg="1"
                decoding="async"
                height="100"
                loading="lazy"
                src="/_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75"
                srcset="/_next/image?url=%2Fsome-image-url1.jpg&w=128&q=75 1x, /_next/image?url=%2Fsome-image-url1.jpg&w=256&q=75 2x"
                style="color: transparent; height: 100%; width: 100%;"
                width="100"
              />
              <button
                class="custom-section-item__play-button"
                data-testid="custom-section-item-play-button"
              >
                <span
                  aria-label="caret-right"
                  class="anticon anticon-caret-right"
                  role="img"
                >
                  <svg
                    aria-hidden="true"
                    data-icon="caret-right"
                    fill="currentColor"
                    focusable="false"
                    height="1em"
                    viewBox="0 0 1024 1024"
                    width="1em"
                  >
                    <path
                      d="M715.8 493.5L335 165.1c-14.2-12.2-35-1.2-35 18.5v656.8c0 19.7 20.8 30.7 35 18.5l380.8-328.4c10.9-9.4 10.9-27.6 0-37z"
                    />
                  </svg>
                </span>
              </button>
            </div>
            <div
              class="music-track-informations"
            >
              <div
                data-testid="music-track-informations-track-name-tooltip"
                placement="right"
              >
                <h4
                  class="music-track-informations__text"
                  data-testid="music-track-informations-track-name"
                >
                  TestTrack
                </h4>
                <span>
                  TestTrack
                </span>
              </div>
              <div
                data-testid="music-track-informations-artist-name-tooltip"
                placement="right"
              >
                <a
                  class="music-track-informations__text music-track-informations__artist"
                  href="/artist/test_artist"
                >
                  <span
                    data-testid="music-track-informations-artist-name"
                  >
                    Test_Artist
                  </span>
                </a>
                <span>
                  Test_Artist
                </span>
              </div>
            </div>
          </div>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
